src/uct/ib/rdmacm/rdmacm_iface.c

```diff
@@ -396,8 +443,25 @@ static UCS_CLASS_INIT_FUNC(uct_rdmacm_iface_t, uct_md_h md, uct_worker_h worker,
 
 static UCS_CLASS_CLEANUP_FUNC(uct_rdmacm_iface_t)
 {
+    uct_rdmacm_cm_id_t *rdmacm_cm_id;
+
     ucs_async_remove_handler(self->event_ch->fd, 1);
-    rdma_destroy_id(self->cm_id);
+    if (self->is_server) {
```

Comment:
can initialize the lists even for client interface, and the code below which goes over the lists will do nothing. and will remove this if for simplicity